models/migrations/v1_22/v285.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_22 //nolint
+
+import (
+	"code.gitea.io/gitea/modules/container"
+
+	"xorm.io/xorm"
+)
+
+func FixMissingAdminTeamUnitRecords(x *xorm.Engine) error {
+	type UnitType int
+	type AccessMode int
+
+	type Team struct {
+		ID         int64      `xorm:"pk autoincr"`
+		OrgID      int64      `xorm:"INDEX"`
+		AccessMode AccessMode `xorm:"'authorize'"`
+	}
+
+	type TeamUnit struct {
+		ID         int64    `xorm:"pk autoincr"`
+		OrgID      int64    `xorm:"INDEX"`
+		TeamID     int64    `xorm:"UNIQUE(s)"`
+		Type       UnitType `xorm:"UNIQUE(s)"`
+		AccessMode AccessMode
+	}
+
+	const (
+		// AccessModeRead read access
+		AccessModeRead = 1
+		// AccessModeAdmin admin access
+		AccessModeAdmin = 3
+
+		// Unit Type
+		TypeInvalid         UnitType = iota // 0 invalid
+		TypeCode                            // 1 code
+		TypeIssues                          // 2 issues
+		TypePullRequests                    // 3 PRs
+		TypeReleases                        // 4 Releases
+		TypeWiki                            // 5 Wiki
+		TypeExternalWiki                    // 6 ExternalWiki
+		TypeExternalTracker                 // 7 ExternalTracker
+		TypeProjects                        // 8 Kanban board
+		TypePackages                        // 9 Packages
+		TypeActions                         // 10 Actions
+	)
+
+	AllRepoUnitTypes := []UnitType{
+		TypeCode,
+		TypeIssues,
+		TypePullRequests,
+		TypeReleases,
+		TypeWiki,
+		TypeExternalWiki,
+		TypeExternalTracker,
+		TypeProjects,
+		TypePackages,
+		TypeActions,
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	// find all admin teams
+	teams := make([]*Team, 0)
+	err := sess.Where("team.authorize = ?", AccessModeAdmin).Find(&teams)
```

Comment:
It may consume much memory for a big instance.

Reply:
Done.